File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/common/SlettOgOpprettResultat.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.common;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class SlettOgOpprettResultat {

    private int antallRadSlettet;
    private int antallRadOpprettet;

    public void add(SlettOgOpprettResultat resultat) {
        antallRadOpprettet = antallRadOpprettet + resultat.getAntallRadOpprettet();
        antallRadSlettet = antallRadSlettet + resultat.getAntallRadSlettet();
    }

    public static SlettOgOpprettResultat tomtResultat() {
        return new SlettOgOpprettResultat(0, 0);
    }
}

File: src/main/kotlin/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/domenemodeller/SlettOgOpprettResultat.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domenemodeller

data class SlettOgOpprettResultat(var antallRadSlettet: Int = 0, var antallRadOpprettet: Int = 0) {
    fun add(resultat: SlettOgOpprettResultat) {
        antallRadOpprettet += resultat.antallRadOpprettet
        antallRadSlettet += resultat.antallRadSlettet
    }

    companion object {

        fun tomtResultat(): SlettOgOpprettResultat {
            return SlettOgOpprettResultat(0, 0)
        }
    }
}
